js/data/create_bucket.js

```diff
@@ -19,6 +19,21 @@ function createBucket(layer, buffers, z, overscaling, collisionDebug) {
         layout[k] = values[k].calculate(z, fakeZoomHistory);
     }
 
+    if (layer.type === 'symbol') {
+        // To reduce the number of labels that jump around when zooming we need
+        // to use a text-size value that is the same for all zoom levels.
+        // This calculates text-size at a high zoom level so that all tiles can
+        // use the same value when calculating anchor positions.
+        if (values['text-size']) {
+            layout['text-max-size'] = values['text-size'].calculate(18, fakeZoomHistory);
+            layout['text-size'] = values['text-size'].calculate(z + 1, fakeZoomHistory);
+        }
+        if (values['icon-size']) {
+            layout['icon-max-size'] = values['icon-size'].calculate(18, fakeZoomHistory);
+            layout['icon-size'] = values['icon-size'].calculate(z + 1, fakeZoomHistory);
+        }
```

Comment:
hmm, yeah I didn't realize that. Thanks for flagging @nickidlugash and @lucaswoj 

There are two parts here.

`.calculate(18)`

This value is used to calculate the label length that is used to calculate the initial offset. This means that at lower zoom levels labels may be offset a bit further from the beginning of the line than they were before. You're right, this is different than before. The differences don't seem to be bad. They're just different:

old:
<img width="1280" alt="screen shot 2015-08-13 at 11 49 50 am" src="https://cloud.githubusercontent.com/assets/1421652/9246041/89d23666-41b1-11e5-95bb-ca87773685ab.png">
<img width="1280" alt="screen shot 2015-08-13 at 11 49 41 am" src="https://cloud.githubusercontent.com/assets/1421652/9246040/89d21f5a-41b1-11e5-83a3-d38a26b11366.png">

I think this is ok

`.calculate(z + 1)`

This is the size used to make the collision boxes. The boxes are a bit bigger now that we use z + 1 instead of just z. This means slightly lower density but also prevents some collisions. It is the same idea @nickidlugash suggested here: https://github.com/mapbox/app/issues/492#issuecomment-74004095

In most cases this won't have a big effect since font size doesn't grow much within a single zoom. When it does grow a lot within a single zoom this will help prevent collisions.